src/test/test.cpp

```diff
@@ -151,15 +152,30 @@ test(int argc, char* const* argv, el::Level ll,
     LOG(INFO) << "Testing stellar-core " << STELLAR_CORE_VERSION;
     LOG(INFO) << "Logging to " << cfg.LOG_FILE_PATH;
 
-    int r = Catch::Session().run(argc, argv);
+    using namespace Catch;
+    Session session{};
+    auto r =
+        session.applyCommandLine(argc, argv, Session::OnUnusedOptions::Ignore);
+    if (r != 0)
+        return r;
+
+    auto unusedTokens = session.unusedTokens();
+    using namespace Clara;
+    gTestAllVersions =
+        std::find_if(std::begin(unusedTokens), std::end(unusedTokens),
```

Comment:
I'd rather have an explicit loop over `unusedToken` and error if we don't recognize one of them

Reply:
Updated.

Note that Catch 2.0 has much better API for adding own command line arguments - https://github.com/catchorg/Catch2/blob/master/docs/own-main.md#adding-your-own-command-line-options

Reply:
cool. Ah yes 2.0 seems promising, but it won't be cheap. I see that they have a bunch of breaking changes like removed `Catch::toString` . We can probably consider switching to it in a few months when they have hammered out the bugs.